internal/pkg/policy/policy_output.go

```diff
@@ -229,6 +233,35 @@ func (p *Output) prepareElasticsearch(
 		output.PermissionsHash = p.Role.Sha2 // for the sake of consistency
 	}
 
+	if outputMap.GetMap(p.Name).GetString("type") == OutputTypeRemoteElasticsearch {
+		om, _ := outputMap.GetMap(p.Name).Marshal()
+		var outputObj map[string]any
+		json.Unmarshal(om, &outputObj)
+		hosts := outputObj["hosts"].([]interface{})
+		outputAPIKey, err :=
+			generateRemoteOutputAPIKey(ctx, agent.Id, p, []string{hosts[0].(string)})
+		if err != nil {
+			return fmt.Errorf("failed generate output API key: %w", err)
+		}
+
+		// TODO update agent doc like above
+		// TODO what about update flow?
+
+		output.APIKey = outputAPIKey.Agent()
```

Comment:
@AndersonQ I saw your comment above, I think we have to update the agent doc with the remote api key. Is there any problem with doing that in the current data model? Or do we have to introduce new fields to store these?

https://github.com/elastic/fleet-server/blame/44b6985dbc49b09bc993bf4c98d593a74bd12bdd/internal/pkg/policy/policy_output.go#L191-L193

Reply:
I kinda take it back. If `prepareElasticsearch` is being called for the fleet-ES and the remote-ES it should just work without any change.

Reply:
The difference in remote ES output is that the service_token comes from the output config, not configured in fleet server itself.